src/apphosting/config.ts

```diff
@@ -54,7 +59,41 @@ export function yamlPath(cwd: string): string | null {
     }
     dir = parent;
   }
-  return resolve(dir, "apphosting.yaml");
+
+  return dir;
+}
+
+/**
+ * Lists absolute paths for `apphosting.*.yaml` configs at project root, if
+ * a project root cannot be found it returns files from cwd.
+ */
+export function discoverConfigsAtProjectRoot(cwd: string): string[] | null {
+  const projectRoot = discoverProjectRoot(cwd);
+  if (!projectRoot) {
+    const configsInCwd = listAppHostingFilesInPath(cwd);
+    if (configsInCwd.length === 0) {
+      return null;
+    }
+
+    return configsInCwd;
+  }
+
+  const configsAtRoot = listAppHostingFilesInPath(projectRoot);
+  if (configsAtRoot.length === 0) {
+    return null;
```

Comment:
why null and not an empty list?

Reply:
It just makes call site cleaner so we don't need to check for empty array there. But looking at this again I can see call sites not knowing this and checking for both null and empty array. Will just go with empty array.